lib/streamlit/connections/snowpark_connection.py

```diff
@@ -28,14 +28,15 @@
 import pandas as pd
 
 from streamlit.connections import ExperimentalBaseConnection
+from streamlit.connections.util import merge_dicts
 from streamlit.errors import StreamlitAPIException
 from streamlit.runtime.caching import cache_data
 
 if TYPE_CHECKING:
     from snowflake.snowpark.session import Session  # type: ignore
 
 
-_REQUIRED_CONNECTION_PARAMS = {"account", "user"}
+_REQUIRED_CONNECTION_PARAMS = {"account", "user", "password"}
```

Comment:
Password is not required if you use SSO (which is quite common)

Reply:
Fixed!